### PR TITLE
Resolves #883. Look in home directory (after pwd) for '.psci'

### LINKED ISSUE #883: Look for a .psci file in the user's home directory
The file in the current directory takes precedence if it exists.

### PR DESCRIPTION
**Pull request by mjgpy3**
Resolves #883. Tested on Linux, I don't have the hardware to test elsewhere.

### CODE REVIEW

File: psci/PSCi.hs [hunk 1]
```diff
@@ -486,18 +486,29 @@ whenFileExists filePath f = do
...
-  exists <- doesFileExist configFile
-  if exists
-  then do
-    ls <- lines <$> readFile configFile
+  localConfigPath <- configFileAt getCurrentDirectory
+  localExists <- doesFileExist localConfigPath
+
+  homeConfigPath <- configFileAt getHomeDirectory
+  homeExists <- doesFileExist homeConfigPath
+  case (localExists, homeExists) of
```
Comment (paf31) [inline psci/PSCi.hs]: This looks like it might be a job for `ExceptT` and `<|>`?

Comment (mjgpy3) [inline psci/PSCi.hs]: Thanks, I'll look into those! I've never used those before, so I'm really grateful for the suggestion :+1:

Comment (paf31) [inline psci/PSCi.hs]: Actually, `MaybeT` would probably do the job, since we don't need an error message.

Comment (mjgpy3) [inline psci/PSCi.hs]: Okay, thanks again!

Comment (mjgpy3) [inline psci/PSCi.hs]: Do we want to load both ($PWD/.psci and $HOME/.psci) if they exist? I assumed not in the above implementation.

**Comment by mjgpy3** [comment]
@paf31 latest push uses `MaybeT`. Awesome suggestion, worked like a charm! If you can spot any further optimizations, let me know. Tested again on my Linux machine.

**Comment by paf31** [comment]
Looks nice! The only thing I would change is to move `readCommands` inside the `where` block, since that's the only place it's used, right?

**Comment by mjgpy3** [comment]
Good catch! Latest push fixes that (I was trying to minimize the git diff but visibility, or lack thereof, is more important)

**Comment by paf31** [comment]
Looks good, thanks! Will merge once Travis is done.

**Comment by mjgpy3** [comment]
Thanks for the review @paf31. I am learning a lot from these PRs :+1:

**Comment by paf31** [comment]
:zap:

**Pull request merged**